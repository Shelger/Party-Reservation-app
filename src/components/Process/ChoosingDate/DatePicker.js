/**
 * @param value     the date users choose
 * 
 */


import React from 'react'
import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
import { DesktopDatePicker } from '@mui/x-date-pickers/DesktopDatePicker';
import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
import TextField from '@mui/material/TextField';
import TimeAvailable from './TimeAvailable';


function DatePicker() {
    const [value, setValue] = React.useState(new Date('2022-01-03T21:11:54'));
    
    const handleChange = (newValue) => {
        setValue(newValue);
    };


    // fetch post get
    return (
        <div style={{margin: 'auto'}}>
            <LocalizationProvider dateAdapter={AdapterDateFns}>
                <DesktopDatePicker
                    label="Date"
                    inputFormat="MM/dd/yyyy"
                    value={value}
                    onChange={handleChange}
                    renderInput={(params) => <TextField {...params} />}
                />
                <TimeAvailable weekday={value.getDay()} />
            </LocalizationProvider>
        </div>
    )
}

export default DatePicker
