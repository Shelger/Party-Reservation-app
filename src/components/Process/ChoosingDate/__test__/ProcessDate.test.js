import React from "react";
import DatePicker from "../DatePicker";
import TestRenderer from 'react-test-renderer'; 
import {cleanup} from "@testing-library/react";

afterEach(cleanup);

it("match ProcessDate", () => {
    const datePicker = TestRenderer.create(<DatePicker />).toJSON();
    expect(datePicker).toMatchSnapshot();
})