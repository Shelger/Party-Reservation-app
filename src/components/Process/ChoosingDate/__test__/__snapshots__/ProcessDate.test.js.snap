// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match ProcessDate 1`] = `
<div
  style={
    Object {
      "margin": "auto",
    }
  }
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={false}
      htmlFor=":r0:"
      id=":r0:-label"
    >
      Date
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="mm/dd/yyyy"
        required={false}
        type="tel"
        value="01/03/2022"
      />
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
      >
        <button
          aria-label="Choose date, selected date is Jan 3, 2022"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CalendarIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1ftyaf0"
        >
          <span>
            Date
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <ul
    className="MuiList-root MuiList-padding MuiList-dense css-1mcnwpj-MuiList-root"
  />
</div>
`;
